### PR TITLE
Increasing test coverage for tristate and delay

### PR DESCRIPTION
**Pull request by Vikram9866**
In call to super changed function to object `Tristate` replaced by `_Tristate`
Instead of
    `super(Tristate, self).__init__(val)`
It should be
   `super(_Tristate, self).__init__(val)`

Reasons:
  Tristate is a function and _Tristate is a class.
  super function is defined as :
  `super(type, obj)` -> bound super object; requires `isinstance(obj, type)`
  `super(type)` -> unbound super object
  `super(type, type2)` -> bound super object; requires `issubclass(type2, type)`

 and `super(Tristate, self)` does not lie in the above mentioned syntax.

Initial test coverage:
>|Name|Stmts|Miss|Cover|
>|---|---|---|---|
>|myhdl\_delay.py|9|1|89%|
>|myhdl\_tristate.py|55|36|35%|

Increased test coverage:
>|Name|Stmts|Miss|Cover|
>|---|---|---|---|
>|myhdl/_delay.py|9|0|100%|
>|myhdl/_tristate.py|55|10|82%|

### CODE REVIEW

File: myhdl/_tristate.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ class _Tristate(_Signal):
             
     def __init__(self, val):
         self._drivers = []
-        super(Tristate, self).__init__(val)
+        super(_Tristate, self).__init__(val)
```
Comment (jck) [inline myhdl/_tristate.py]: Good catch, I think this was definitely a bug.

**Comment by jck** [comment]
I don't understand the point of these delay tests, can you explain?

**Comment by Vikram9866** [comment]
I added delay test to increase test coverage for `_delay.py `. Its a very trivial check to make, but I assume having `100%` test coverage would be better.

**Comment by jck** [comment]
I mean, what exactly is it testing for?

**Comment by Vikram9866** [comment]
There was no test case checking the behavior of `delay` function for non-integer and negative delay values. So I made a separate test file which verifies the expected behavior.

`delay` function present in `_delay.py` should raise a `TypeError` for negative and non-integer cases.

**Pull request closed without merging**